### PR TITLE
dep: update TiDB and parser

### PR DESCRIPTION
**Pull request by lance6716**
Signed-off-by: lance6716 <lance6716@gmail.com>

### What problem does this PR solve?

Issue Number: ref #596 

### What is changed and how it works?

update parser to include https://github.com/pingcap/tidb/pull/33212

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
   compiled sync-diff and passed a DM integration test locally https://github.com/pingcap/tiflow/pull/4949

Code changes

Side effects

Related changes

### CODE REVIEW

**Comment by lance6716** [comment]
/unhold

local test passed

File: go.mod1 [hunk 1]
```diff
@@ -32,6 +32,9 @@ require (
 
 replace github.com/golang/lint => golang.org/x/lint v0.0.0-20190930215403-16217165b5de
 
+// FIXME: Change to master TiDB after upgrade TiDB.
+replace github.com/pingcap/tidb => github.com/oh-my-tidb/tidb v1.1.0-beta.0.20220218042203-2e273f3f990d
```
Comment (amyangfei) [inline go.mod1]: Is this replace expected?

**Review by amyangfei — APPROVED** [review]

**Pull request merged**